plugins/catalog-backend/src/stitching/buildEntitySearch.ts

```diff
@@ -134,13 +134,22 @@ export function mapToRows(input: Kv[], entityId: string): DbSearchRow[] {
   for (const { key: rawKey, value: rawValue } of input) {
     const key = rawKey.toLocaleLowerCase('en-US');
     if (rawValue === undefined || rawValue === null) {
-      result.push({ entity_id: entityId, key, value: null });
+      result.push({
+        entity_id: entityId,
+        key,
+        original_value: null,
+        value: null,
+      });
     } else {
       const value = String(rawValue).toLocaleLowerCase('en-US');
       if (key.length <= MAX_KEY_LENGTH) {
         result.push({
           entity_id: entityId,
           key,
+          original_value:
+            String(rawValue).length <= MAX_VALUE_LENGTH
```

Comment:
You can just check value.length, it's the same.

I see that this set of if statements is already a bit odd before your PR. Seems to me, that we should check `if (key.length <= MAX_KEY_LENGTH)` FIRST and do absolutely nothing if it is too long. THEN check, if the value is too long, then just do an insertion where original_value and value are null, otherwise do the "base case" regular insert of the proper values